asreview/project.py

```diff
@@ -59,60 +63,126 @@ class ProjectNotFoundError(Exception):
     pass
 
 
-def get_project_path(project_id, asreview_dir=None):
+# !=============================================
+# @TODO: I HATE THE FACT THAT I AM TALKING ABOUT
+# Users in this module, but I need a quick fix
+# for now
+# !=============================================
+
+
+def get_project_path(project_id, user=None, asreview_dir=None):
     """Get the project directory.
 
     Arguments
     ---------
     project_id: str
         The id of the current project.
+    user: User
+        User account object.
     """
 
     if asreview_dir is None:
         asreview_dir = asreview_path()
 
-    return Path(asreview_dir, project_id)
+    if isinstance(user, User) and isinstance(user.id, int):
+        # if we do authentication, then the project must be 
+        # registered in the database
+        project_from_db = Project.query.filter(
+            Project.project_id == project_id
+        ).one_or_none()
 
+        # project exists and user -is- owner OR this is a new project
+        if (project_from_db and user == project_from_db.owner) or (
+            project_from_db == None
+        ):
 
-def project_from_id(f):
+            folder_id = f"{user.id}_{project_id}"
 
-    @wraps(f)
-    def decorated_function(project_id, *args, **kwargs):
+        # project exists but user is a collaborator
+        elif project_from_db and user in project_from_db.collaborators:
+            folder_id = f"{project_from_db.owner_id}_{project_id}"
 
-        project_path = get_project_path(project_id)
-        project = ASReviewProject(project_path, project_id=project_id)
-        return f(project, *args, **kwargs)
+        # default to project_id
+        else:
+            folder_id = project_id
+            
+    else:
+        folder_id = project_id
+
+    project_folder = uuid5(NAMESPACE_URL, folder_id).hex
 
-    return decorated_function
+    return Path(asreview_dir, project_folder)
 
 
-def list_asreview_projects():
-    """List the projects in the asreview path"""
+def project_from_id(user):
+    """Decorator function that takes a user account as parameter,
+    the user account is used to get the correct sub folder in which
+    the projects is
+    """
+
+    def decorate(f):
+        @wraps(f)
+        def decorated_function(project_id, *args, **kwargs):
+            project_path = get_project_path(project_id, user)
+            project = ASReviewProject(project_path, project_id=project_id)
+            return f(project, *args, **kwargs)
+
+        return decorated_function
+
+    return decorate
+
+
+# !=============================================
+# @TODO: Another thing that needs consideration:
+# I need ownership id's to avoid collaborators inviting
+# new collaborators. That means I'd like to alter
+# the ASReviewProject class -> adding an ownership
+# attribute. I could add it on the class itself, but
+# this is only needed in the authenticated version.
+# Instead I will add it dynamically in this function...
+# Feedback is required.
```

Comment:
I agree that this is a bit ugly. I think I would prefer a `project.owner_id` attribute which can be None. Then you are always sure that the attribute is there. But maybe we should first work towards merging with the asreview/auth-v1 branch and only after try to change this. @J535D165 Do you have an opinion?